aicsimageio/dimensions.py

```diff
@@ -142,3 +142,10 @@ def __getitem__(self, key: Union[str, Sequence[str]]) -> Tuple[int, ...]:
             raise TypeError(
                 f"Key must be a string or list of strings but got type {type(key)}"
             )
+
+    def __setattr__(self, __name: str, __value: int) -> None:
+        super().__setattr__(__name, __value)
+
+    def __getattr__(self, __name: str) -> int:
+        # TODO: Py310 __match_args__ for better typing
+        return super().__getattribute__(__name)
```

Comment:
So the above code is the answer to nearly all of the type ignores?  Am I understanding that right?

Reply:
The above code is the answer to _many_ type ignores but not all of the ones cleared in this PR. This change clears any of the type ignores related to `dims.` stuff.

The other type ignores that are cleared seem to be because upstream libraries got better typing (the main one I am thinking of is that xarray coords now properly accepts a `Dict` type).

This PR includes both the dimensions type fixes and then just also happens to clear up type ignores because upstreams got better typing too.